p2p/service.go

```diff
@@ -0,0 +1,71 @@
+package p2p
+
+import (
+	"context"
+
+	"github.com/pastelnetwork/gonode/common/log"
+	"github.com/pastelnetwork/gonode/p2p/kademlia"
+	"github.com/pastelnetwork/gonode/p2p/kademlia/dao/mem"
+)
+
+const (
+	logPrefix = "p2p"
+)
+
+// Service represents the p2p service.
+type Service struct {
+	dht DHT
+}
+
+// DHT represents the methods by which a library consumer interacts with a DHT.
+type DHT interface {
+	Store(ctx context.Context, data []byte) (id string, err error)
+	Get(ctx context.Context, key string) (data []byte, found bool, err error)
+	CreateSocket() error
+	Listen(ctx context.Context) error
+	GetNetworkAddr() string
+	Bootstrap(ctx context.Context) error
+	UseStun() bool
+	Disconnect() error
+}
+
+// Run starts the DHT service
+func (service *Service) Run(ctx context.Context) error {
+	ctx = log.ContextWithPrefix(ctx, logPrefix)
+
+	err := service.dht.CreateSocket()
+	if err != nil {
+		panic(err)
```

Comment:
Don't forget to remove `panic`
![image](https://user-images.githubusercontent.com/62853952/119413163-c4d16a00-bcf5-11eb-84a5-2834ecd96d72.png)

Reply:
removed all panics